### ISSUE TITLE
NAT

### ISSUE THREAD

**Issue by ddominet** [cpriest/udm-patches#3]
Hi, Script works great with one small problem.
It seems that UDM still NATs the ip recieved from VPN server. 
I can ping any device connected to VPN Server from my local network behing UDM but I can't ping any devices in local network from VPN-SERVER, even UDMs IPs in the local network. It looks like NAT to me.
Have you encountered similar dificulties ?

**Comment by cpriest** [comment]
That would make sense, my setup uses NAT on the UDM because there are no reverse routes w/ my OpenVPN setup.  With any NAT setup any ingress packets from the VPN connection would be dropped like it was firewalled because iptabels doesn't know what to do with it.

If you comment out the lines that add/remove to the -t NAT table, that should do it if your remote end has reverse routes.

**Comment by ddominet** [comment]
Might I ask about one more thing. I did as you said, but unfortunately now only udm itself can ping the remote server. Remote end has all the routes it needs. I tried tcpdump to figure out traffic, but it seems that UDM sends out packets but they don't arrive. Same behaviour with the remote end as the sending side. I wondered if this is a problem related to iptables. 

I tried adding those rules: 
iptables -A UBIOS_INPUT_USER_HOOK -j UBIOS_LAN_LOCAL_USER -i tun1  
iptables -A UBIOS_FORWARD_IN_USER -j UBIOS_LAN_IN_USER -i tun1 
iptables -A UBIOS_FORWARD_OUT_USER -j UBIOS_LAN_OUT_USER -o tun1 
iptables -I UBIOS_LAN_IN_USER -j RETURN -s 10.8.0.0/24
iptables -I UBIOS_LAN_OUT_USER -j RETURN -d 10.8.0.0/24

unfortunately without much success.

What do you think might be causing this? There is no problem while using masquerade, but it severely limits the use-case. 
Thanks for your support,
Dominic

**Comment by ddominet** [comment]
I can add that I can't ping from UDM ip's in different subnets. It might be obvious, but I don't want to leave that out.

**Comment by ddominet** [comment]
Never mind 😅, I did thorough breakdown of this scenario in lab environment and this script was not an issue. Good day to you

**Comment by cpriest** [comment]
Sorry I hadn't responded timely to this.  I know just enough about networking to be dangerous, it took me quite some time to figure out the problem I was having (which was not having the NAT going on); I knew that wouldn't work for everyone but that's the way our company has it setup.

If there's any contribution you could make, even if the solution to your problem in this ticket, it would only help others.

**Comment by ddominet** [comment]
I'm using this at home. In my setup which used openvpn-install script by angristan 
*/i know I'm lazy 😄 /*, for a vps server. I found out that the problem wasn't with the reverse routes or udm's firewall but in fact was happening due to NAT rules that were added with the script on openvpn server. Due to the fact that openvpn on UDM doesn't have hardware acceleration I'll experiment with other solutions such as nebula vpn mesh. I'll write a script on that soon.
